webvtt.go

```diff
@@ -36,8 +36,8 @@ var (
 	bytesWebVTTTimeBoundariesSeparator = []byte(webvttTimeBoundariesSeparator)
 	webVTTRegexpInlineTimestamp        = regexp.MustCompile(`<((?:\d{2,}:)?\d{2}:\d{2}\.\d{3})>`)
 	webVTTRegexpTag                    = regexp.MustCompile(`(</*\s*([^\.\s]+)(\.[^\s/]*)*\s*([^/]*)\s*/*>)`)
-	webVTTEscaper                      = strings.NewReplacer("&", "&amp;", "<", "&lt;")
-	webVTTUnescaper                    = strings.NewReplacer("&amp;", "&", "&lt;", "<")
+	webVTTEscaper                      = strings.NewReplacer("astisub.amp", "&")
```

Comment:
I don't really like the idea of using this `astisub.amp` string to mess with html escape/unescape 🤔 

Wouldn't adding `"\u00A0", "&nbsp;"` to `htmlEscaper` and `"&nbsp;", "\u00A0""` to `htmlUnescaper` do the trick? 🤔 ([\u00A0](https://www.compart.com/en/unicode/U+00A0))

Reply:
This solves for all html entities (e.g. `&euro;` `&quot;` ...) rather than keeping a short list of acceptable entities. I tried just using go's `html.EscapeString`, however that leads to converting everything in the output. So, things like single quotes are forced to `&#39;` double quotes go to `&#34;`. Not sure if that is acceptable for this library.

Reply:
The problem is that when converting to `ssa`, `stl` or `teletext` there will be `astisub.amp` as well in the output. What is parsed from any subtitle format and written in the `Subtitles` struct should not be format-specific and should be usable by all formats. Therefore we need to transform the html entities into "neutral" characters that express themselves properly and are usable by all formats: `&amp;` can become `&`, `&lt;` `<`, `&euro;` `€`, `&quot;` `"`, etc. Unfortunately, I don't see another way than having a short list of acceptable entities 🤔

Reply:
Understood. I updated my changes back to using a short list of acceptable entities with nbsp added. I might look into a larger solution for this later.